### PR TITLE
Move queue management to dispatcher

### PR DESCRIPTION
**Pull request by michel-laterman**
## What does this PR do?

Move queue management actions to the dispatcher from the fleet-server in order to help with future work to add a retry mechanism. Add a PersistedQueue type which wrap the ActionQueue to make persisting the queue simpler for the consumer.

Queuing was added as part of https://github.com/elastic/elastic-agent/pull/419

## Why is it important?

Retry actions on edge will be implemented in the dispatcher by interacting with the queue. Moving all of the control flow for the queue to the dispatcher makes it more straightforward to understand and debug

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~I have made corresponding changes to the documentation~~
- ~~I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- ~~I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

## How to test this PR locally

Follow guide in #419 

## Related issues

- Relates https://github.com/elastic/elastic-agent/issues/778

### CODE REVIEW

**Comment by michel-laterman** [comment]
rebasing on github is mesy I'll make a new PR for v2 without any of the garbage

**Comment by michel-laterman** [comment]
Uses fixes from the v2 attempt: https://github.com/elastic/elastic-agent/pull/1109

File: internal/pkg/agent/application/gateway/fleet/fleet_gateway.go [hunk 1]
```diff
@@ -175,28 +161,12 @@ func (f *fleetGateway) worker() {
...
 				continue
 			}
-
-			actions := f.queueScheduledActions(resp.Actions)
-			actions, err = f.dispatchCancelActions(actions)
-			if err != nil {
-				f.log.Error(err.Error())
+			actions := make([]fleetapi.Action, len(resp.Actions))
+			for idx, a := range resp.Actions {
+				actions[idx] = a
```
Comment (blakerouse) [inline internal/pkg/agent/application/gateway/fleet/fleet_gateway.go:166]: Instead of this forloop copy should work.

```
copy(actions, resp.Actions)
```

**Review by blakerouse — APPROVED** [review]
Change looks good.

File: internal/pkg/agent/application/gateway/fleet/fleet_gateway.go (hunk 1 above)
Comment (michel-laterman) [reply internal/pkg/agent/application/gateway/fleet/fleet_gateway.go:166]: ok, loop was the original implementation. i'll change to copy

Comment (michel-laterman) [reply internal/pkg/agent/application/gateway/fleet/fleet_gateway.go:166]: I removed the change as the build was failing

**Comment by cmacknz** [comment]
The Mac build being broken is a new issue, main just updated to switch to Github actions which should help. There is an open investigation with the CI team about why the Mac workers aren't reliable.

**Comment by michel-laterman** [comment]
Closing, v2 PR was merged as the feature has been pushed to 8.6

**Pull request closed without merging**